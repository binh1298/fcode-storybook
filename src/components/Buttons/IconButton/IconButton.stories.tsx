import { Add as AddIcon, Edit as EditIcon } from "@material-ui/icons";

import IconButton, { FabBaseProps } from ".";

import { Story } from "@storybook/react";

export default {
    title: "Components/IconButton",
    component: IconButton,
};

const Template: Story<FabBaseProps> = (args) => <IconButton {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    children: <AddIcon />,
    color: "primary",
    variant: "round",
};

export const Secondary = Template.bind({});
Secondary.args = {
    children: <EditIcon />,
    color: "secondary",
    variant: "round",
};
