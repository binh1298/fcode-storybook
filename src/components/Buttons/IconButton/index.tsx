import { Fab as MaterialFab, FabProps as MaterialFabProps } from "@material-ui/core";

export interface FabBaseProps extends MaterialFabProps {}

const IconButton = (props: FabBaseProps) => {
    return <MaterialFab {...props} />;
};

export default IconButton;
