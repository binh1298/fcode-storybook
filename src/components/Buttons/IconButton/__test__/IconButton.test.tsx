import IconButton, { FabBaseProps } from "../";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<ButtonBase />", () => {
    let wrapper: RenderResult;

    const props: FabBaseProps = {
        children: <div>Test</div>,
    };

    beforeEach(() => {
        wrapper = render(<IconButton {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
