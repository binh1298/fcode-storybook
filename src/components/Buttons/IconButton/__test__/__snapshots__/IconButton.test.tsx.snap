// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBase /> should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiFab-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <div>
        Test
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
